engine/renderer/private/gpu_scene.cpp

```diff
@@ -149,6 +161,29 @@ void GPUScene::UpdateDirectionalLightData(SceneData& scene, uint32_t frameIndex)
         directionalLightIsSet = true;
     }
 }
+void GPUScene::UpdatePointLightData(PointLightArray& pointLightArray, MAYBE_UNUSED uint32_t frameIndex)
+{
+    auto pointLightView = _ecs.GetRegistry().view<PointLightComponent, TransformComponent>();
+    uint32_t pointLightCount = 0;
+
+    for (const auto& [entity, pointLightComponent, transformComponent] : pointLightView.each())
+    {
+        if (pointLightCount >= MAX_POINT_LIGHTS)
+        {
+            bblog::warn("Reached the limit of point lights available");
+            break;
+        }
+
+        PointLightData& pointLightData = pointLightArray.lights[pointLightCount];
+        pointLightData.position = glm::vec4(TransformHelpers::GetLocalPosition(transformComponent), 1.0f);
```

Comment:
Local position? Why not world position, since this would break hierarchy, right? World position is free anyway since we have the world matrix cached

Reply:
How would I do this? Because the returns a mat4, are there any examples that use this?